tools/mo/openvino/tools/mo/ops/gather.py

```diff
@@ -89,6 +89,7 @@ def infer(node: Node):
         data_value = node.in_port(0).data.get_value()
         indices_value = node.in_port(1).data.get_value()
         if data_value is not None and indices_value is not None and is_fully_defined(indices_value):
+            indices_value = int64_array(indices_value)
```

Comment:
why is this change needed if we care of a type of indices overall?

Reply:
np.ma.take below raise an exception if `indicies_value` has another type. Found this case in the customer model, there was an array with `dtype` = `float`.